### ISSUE TITLE
🪲 Bug Report - Did not collect Advisor recommendations

### ISSUE THREAD

**Issue by tksh164** [Azure/Well-Architected-Reliability-Assessment#57]
## Describe the bug

Start-WARACollector did not collect Advisor recommendations because advisorResourceObj hasn't a right method for the `+` operator.

https://github.com/Azure/Well-Architected-Reliability-Assessment/blob/cb1012ea685388b20956fc8c0134d04964c83ffa/src/modules/wara/scope/scope.psm1#L306

## To Reproduce

1. Run Start-WARACollector with `-ResourceGroups` parameter like the following:

    ```powershell
    Start-WARACollector -TenantID <TENANT_ID> -ResourceGroups <RESOURCE_GROUP_ID>
    ```

## Expected behavior

Advisor recommendations are collected properly even if specify resource groups.

## Screenshots 📷

![Image](https://github.com/user-attachments/assets/2781f243-a966-4c77-aaa2-949fdd3eb14a)

![Image](https://github.com/user-attachments/assets/822945dc-b1dd-4744-88e1-242c2f727cbe)

## Additional context

- n/a

**Comment by kpoineal** [comment]
@tksh164 I can't reproduce this issue. Can you try again?

**Comment by tksh164** [comment]
@kpoineal Yes, it still exists. You need to set `$ErrorActionPreference = 'Stop'` to stop execution when the exception thrown. If not set, the exception log be swept away by many logs after that.

![Image](https://github.com/user-attachments/assets/2225c0f0-9ab1-4744-975b-eb645022c258)

![Image](https://github.com/user-attachments/assets/67acb51e-8351-47d7-b7ee-22404a44a2b8)


The following line:

https://github.com/Azure/Well-Architected-Reliability-Assessment/blob/e95a5926d3170389ed4c09d3ff2f4af669e684a1/src/modules/wara/scope/scope.psm1#L319

It will actually be the following at the exception thrown. This is the reason for the exception, I think.

```powershell
$FilteredResources = $null + $ResourceGroupFilteredResources + $null | Select-Object -Property * -Unique
```

I will investigate it more and fix it.

**NOTE:** We should set `$ErrorActionPreference = 'Stop'` in the wara module.

**Comment by kpoineal** [comment]
My testing returns advisorobjects, but you're right, those shouldn't be $null they should be empty collections in that function.

I'll investigate more tomorrow as well.

**Comment by tksh164** [comment]
@kpoineal `$ResourceGroupFilteredResources` is not collection when the exception thrown.

The following is the minimum repro code.

```powershell
class advisorResourceObj {
    advisorResourceObj() {
    }
}

function func1 {
    $ret = @()
    $ret += [advisorResourceObj]::new()
    return $ret
}

$ar = func1
$ar + $null
```

PowerShell's function does not return an array if the array has only a single element even if the variable is an array in the function. The following is the fixed code. I fixed the same issue with #48 previously.

```powershell
class advisorResourceObj {
    advisorResourceObj() {
    }
}

function func1 {
    $ret = @()
    $ret += [advisorResourceObj]::new()
    return ,$ret  # Added an array operator to always return as an array.
}

$ar = func1
$ar + $null
```
